### ISSUE TITLE
Autumn MVC documentation

### ISSUE THREAD

**Issue by WimpyMistake** [czyzby/gdx-lml#16]
Hello, sorry this is not an issue. Not even a contribution. Just a series of questions/feedback.
But before that, congratulations and thank you for making this amazing framework. Being a Spring MVC user, i was very excited when i discovered this autumn mvc project.
While i was fascinated with tons of features autumn-mvc offers, i am left banging the walls. Because i just don't know how to use this. I have tried mvc-simple example, fiddled around with it, but that just as far i can go. I can do trivial stuff like create some actors in LML, create controllers. But that's it. When i about to start making a game, i just got stuck. For example, how can i use box2d here? How can i spawn actors into LML? How to have hero character? How to do scene2d actions?
I have read your readme files and wiki but not fruitful. In mvc-simple example, I tried to use new skin using `@Skin`, but then can't load with VISUI skin. How can import my own game asset (sprites). For now i rebuild VisUi raw asset and use as skin as stated [here](https://github.com/kotcrab/vis-editor/wiki/VisUI#modifying-skin). I know what annotations/features are available, but just no idea how to use them. And i'm unsure what to put in Service layer.

tldr;
i hope its not too much if i ask some more complete game example. A simple jumper game with box2d will do. Or if box2d is too much, just some simple card game should suffice to demonstrate scene2d usage.

Thank you. Sorry for long post or sounding like complaining. Just my 2 cents.

**Comment by czyzby** [comment]
I completely understand your points, as `Autumn MVC` was basically based on my personal utilities, which I decided made public. It definitely needs more documentation, and you don't sound like complaining at all.

As for your questions:
- To include `gdx-lml-vis` in your project, read [this `README` section](https://github.com/czyzby/gdx-lml/tree/master/lml-vis#including-gdx-lml-vis-in-autumn-mvc).
- When use add a `@ViewStage Stage stage;` field to your `@View[Dialog]`-annotated class, `Stage` instance will be automatically injected when the view is created. Then you can use the standard `Scene2D` API to do whatever you want.
- I think `@Skin` was a simple loading utility - you give at a *path* to your skin you make it load it for you. See the first answer on how to include **VisUI**.
- Read `@View` javadocs. When your annotated class implements `ViewRenderer` interface, you can handle how the view is drawn. This is a utility for this exact kind of problem: you need to render some additional stuff, which cannot be handled by simple GUI mechanism. For example, you can start with drawing your Box2D-related sprites, and then just go with `Stage` drawing for the HUD.
- `@Component`-annotated classes are scanned for in your application and initiated. They can be `@Inject`-ed like any other classes. I usually go with this annotation for service layer components. See [`gdx-autumn-tests`](https://github.com/czyzby/gdx-lml/tree/master/examples/gdx-autumn-tests) for basic usage of core annotations.

Not sure I should link to this, but I made an idle game a *while* back. It was quick, kind of dirty and used some of the earliest Autumn versions. I don't think it's worth maintaining (that's why it wasn't moved to this repository), but you can find it here, if you really want a simple example game: [gdx-idle](https://github.com/czyzby/gdx-autumn-mvc-tests/tree/master/gdx-idle). I was thinking about rewriting **Super Jumper** (the standard LibGDX example) to Autumn MVC, but give me some time - I *really* rewriting code that's not my own. ; )

I don't mind writing some extra documentation for Autumn, so if you have any suggestions, don't hesitate to write it here.

**Comment by WimpyMistake** [comment]
Thank you for your answers and gdx-idle! It surely helping me. I have learnt so many stuff from it. Now i realised that i have been doing things wrongly LOL.
Its okay, take your time with the documentation and sample game. Thanks a lot man.

**Comment by czyzby** [comment]
Sure, create an issue or post here any time you have questions.

By the way, do you have any suggestions for wiki pages? I think about creating a simple application tutorial comparable to [your first LML app](https://github.com/czyzby/gdx-lml/wiki/Your-first-LML-application) - but resulting with SuperJumper instead of some mock-up application. Not sure if that's enough, though.

**Comment by WimpyMistake** [comment]
Oh yes sure. These are what i have in mind for now. They are basically HOW-TOs, some are trivial but i believe would be helpful for beginners. Can start with basics and can add more advance stuff later.

- requirements, if any (which you already have [in your first LML app](https://github.com/czyzby/gdx-lml/wiki/Your-first-LML-application))
- how to create a project (which you already have [in your first LML app](https://github.com/czyzby/gdx-lml/wiki/Your-first-LML-application))
- how to prepare assets (point out usage of texture packer, how to customize VisUI skin, provide some links to external tutorial about skin)
- how to set up the DTD
- creating Root class and how to launch this Root from desktop, android, html (and IOS in future)
- create the configuration class (need to emphasise which config most commonly used, what can be config here, how to a skin, how to use VisUI (or maybe make *VisUI* as default), customise *VisUI*, or use multiple skin)
- create LML and use it with controllers..common option in `@View`..specify what kind of logic should be here (i'm assuming UI logic). this class should implement/extend which class..demonstrate component injection, actor injection, controlling actor from LML
- create service layer for game logic..some common usage example : player control, collision detection, spawning actors, invoking scene2d actions, controlling camera, Box2d integration.
- demonstration of the features/Annotations (for example : saving preferences)

SuperJumper demo should suffice for most people :)
Thank you MJ.

**Comment by czyzby** [comment]
Actually, having looked through LibGDX demos, I think that [The plane that couldn't fly good](https://github.com/badlogic/theplanethatcouldntflygood) has much nicer graphics and should be simple enough for a first project. One could easily extend it with menu, settings, etc.

**Comment by WimpyMistake** [comment]
That looks good too. A flappy bird clone is somewhat popular example also. As long as it has enough info to get people started.

**Comment by czyzby** [comment]
Just wanted you to know that I added a solid base for Autumn MVC applications to the new `gdx-setup`. While it's not that much different than the `simple` example, it should quickly get you started and limit the time you spend on copy-pasting the code from examples projects to your application. It features loading screen, settings dialog and a main menu with custom rendering method.

See https://github.com/czyzby/gdx-setup/issues/7

**Comment by czyzby** [comment]
@WimpyMistake Ask and you shall receive. (Or something like that.) I just included a *massive* Autumn MVC project template, including usage of Box2D, Controllers and VisUI extensions. It helps solving some real life problems like customizing controls and managing full screen. While the Box2D world is rather simple, it's good enough to get you started. See [gdx-setup](https://github.com/czyzby/gdx-setup).

**Comment by WimpyMistake** [comment]
Awesome as usual! Sorry for the late reply. I lost my internet for few weeks. I have checked the sample out, looks neat. Now time to experiment. Will report issues, add suggestions while im on it.
Thanks heaps

**Comment by czyzby** [comment]
I realized Autumn MVC is way too complicated to be widely used, so I'm currently (slowly, but surely) making progress on a much simpler, while still pretty powerful framework. You might want to watch is progress here: https://github.com/czyzby/gdx-lml/issues/26 I'm not dropping Autumn MVC support though, so if you notice any bugs or have any questions, don't hesitate to create issues.

**Comment by czyzby** [comment]
I lack the time and resources to prepare significant documentation for Autumn MVC. To be honest, I doubt there are many users of this (semi-)framework - *LML* seems to be my most commonly used project by far. Existing Autumn example projects, templates from the new `gdx-setup` and nearly full javadocs should be enough to get you started - if you have any problems with any functionalities, please, create a separate issue with your questions and I'll do my best to answer.
